akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -59,38 +59,82 @@ private[akka] final case class TakeWhile[T](p: T ⇒ Boolean, decider: Supervisi
 /**
  * INTERNAL API
  */
-private[akka] final case class DropWhile[T](p: T ⇒ Boolean, decider: Supervision.Decider) extends PushStage[T, T] {
+private[stream] final case class DropWhile[T](p: T ⇒ Boolean) extends GraphStage[FlowShape[T, T]] {
+  val in = Inlet[T]("DropWhile.in")
+  val out = Outlet[T]("DropWhile.out")
+  override val shape = FlowShape(in, out)
+  override def initialAttributes: Attributes = DefaultAttributes.dropWhile
   var taking = false
 
-  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
-    if (taking || !p(elem)) {
-      taking = true
-      ctx.push(elem)
-    } else {
-      ctx.pull()
+  def createLogic(inheritedAttributes: Attributes) = new SupervisionedGraphStageLogic(inheritedAttributes, shape) with InHandler with OutHandler {
+    override def onPush(): Unit = execute(() ⇒ {
+      val el = grab(in)
+      if (taking || !p(el)) {
+        taking = true
+        push(out, el)
+      } else
+        pull(in)
+    })
```

Comment:
Something I personally am unsure about, shouldn't the supervision logic _only_ be for the call to `p` here, with the current impl it would do supervision if the stage threw "pulled twice" etc. which isn't really something that can be resumed/restarted?